perf/perf_lock.py

```diff
@@ -0,0 +1,160 @@
+            self.fail("Command %s failed: %s" % (cmd, details))
+
+    def check_perfdata_file_exist(self):
+        '''
+        Function to verify "perf.data" file generated by
+        "perf record" command.
+
+        ::rtype : True when file got generated by "perf record"
+
+        '''
+        if os.path.exists(self.output_file):
+            if not os.stat(self.output_file).st_size:
+                self.fail("%s sample not captured" % self.output_file)
+            return True
+
+    def get_kernel_version(self):
+        '''
+        Funtion to get kernel version based on distro type
+        '''
+
+        if 'rhel' in self.distro_name:
+            self.kernel_version = self.option + \
+                    " /boot/vmlinuz-" + platform.uname()[2]
+        elif 'SuSE' in self.distro_name:
+            self.kernel_version = self.option + \
+                    " /boot/vmlinux-" + platform.uname()[2]
+        return self.kernel_version
+
+    def test_lock(self):
+        # When input is used for report, then need to pass the file argument
+        # to the same file, record should log the data. So altering record,
+        # report options to have the proper arguments.
+
+        self.output_file = "perf.data"
+
+        # Record command
+
+        record_cmd = "perf lock record -o perf.data -a -v sleep 3 "
+        self.run_cmd(record_cmd)
+
+        # Report command
+        if self.record == 'report':
+            report_cmd = "perf lock %s %s " % (self.record, self.option)
+            if self.option in ["--kallsyms"]:
+                self.option = self.get_kernel_version()
+                report_cmd = "perf lock %s %s -E 3" % (self.record,
+                                                       self.option)
+            if self.check_perfdata_file_exist:
+                self.run_cmd(report_cmd)
+
+        # Info command
+        elif self.record == 'info':
+            report_cmd = "perf lock %s %s" % (self.record, self.option)
+            if self.check_perfdata_file_exist:
+                self.run_cmd(report_cmd)
+
+        # Script command
+        elif self.record == 'script':
+            report_cmd = "perf lock %s %s %s" % (self.record,
+                                                 self.option,
+                                                 self.output_file)
+            if self.check_perfdata_file_exist:
+                self.run_cmd(report_cmd)
+
+        # contention command
+        elif self.record == 'contention':
+            report_cmd = "perf lock %s %s" % (self.record, self.option)
+            if self.option in ["--vmlinux"]:
+                self.option = self.get_kernel_version()
+                report_cmd = "perf lock %s %s" % (self.record, self.option)
+            elif self.option in ["-i"]:
+                report_cmd = "perf lock %s %s %s" % (self.record,
+                                                     self.option,
+                                                     self.output_file)
```

Comment:
All this can be clubbed in a single command. Inside this you can have checks for --kallsyms and --vmlinux. Other than that other parameters should be part of yaml file.
`report_cmd = "perf lock -i %s %s %s" % (self.output_file, self.record, self.option)`